### PR TITLE
Concierge: Remove async loading of the reducer.

### PR DESCRIPTION
**Pull request by kwight**
Concierge currently has its reducer loaded async by the two places that use it: Concierge itself, and `/me/purchases`. Now that it's also being used by My Home (the default landing page for logged-in users), it makes sense to handle it normally in the root reducer (it can also avoid mistakes like mine in #41112 🙃 ).

**Testing Instructions**
* Apply this branch.
* Go to `/me/concierge`, and verify it loads properly, without console errors referring to `Reducer with key 'concierge' is already registered`.
* Go to `/me/purchases`, and verify the same.
* Check the Calypso codebase for any missed instances of async loading for this reducer.

### CODE REVIEW

**Review by gwwar — APPROVED** [review]
Thanks, this tests well for me @kwight, let's get a +1 from @sirbrillig or the payments team before we land.

**Comment by gwwar** [comment]
Note that dserve is having issues pMz3w-bnn-p2

**Comment by sirbrillig** [comment]
Thanks for the ping! Will test very shortly.

**Review by sirbrillig — APPROVED** [review]
Tested and I see no errors! Thanks!

**Comment by sgomes** [comment]
@Automattic/team-calypso 

Hey folks!

Calypso is actually trying to move away from having a monolithic reducer, to a modularised reducer approach. State is one of the largest parts of the codebase, and loading it all on boot is an issue, since each section only needs a comparatively small portion of it.

There's a mechanism in place to allow for this relatively easily, without the custom code that is being replaced here, which required knowledge of which pages use which portions of state. All it takes is having an init file for the portion of state and importing it in every action creator or selector module that touches that part of state. See e.g. see #39261 and p4TIVU-9lM-p2 for more details.

This seems like a good opportunity to move concierge state to the modularised model. I'll prepare a PR and send the review your way!

**Comment by sgomes** [comment]
Opened #41797 with the modularised concierge state.

**Pull request merged**
